mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java

```diff
@@ -0,0 +1,54 @@
+package io.mosip.esignet.mock.identitysystem.validator;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.mosip.esignet.mock.identitysystem.dto.IdentityData;
+
+@Component
+public class IdentityDataValidator implements ConstraintValidator<IdData, IdentityData> {
+
+	@Value("${mosip.esignet.mock.supported-fields}")
+	private String supportedFields;
+
+	@Value("${mosip.esignet.mock.number-of-extra-fields:10}")
+	private String extraFields;

```

Comment:
Lets declare this as int or long directly and can we rename to allowedExtraFieldsCount ?

Reply:
Removed the count as we discussed.